homeassistant/components/climate/generic_thermostat.py

```diff
@@ -30,27 +32,35 @@
 
 DEFAULT_TOLERANCE = 0.3
 DEFAULT_NAME = 'Generic Thermostat'
-DEFAULT_AWAY_TEMP = 16
+DEFAULT_AWAY_TEMP_COOL = 30
+DEFAULT_AWAY_TEMP_HEAT = 16
+DEFAULT_TARGET_TEMP_HIGH = 21
+DEFAULT_TARGET_TEMP_LOW = 18
+DEFAULT_TARGET_TEMP = 20
 
-CONF_HEATER = 'heater'
+CONF_HEATER = 'heater_control'
 CONF_SENSOR = 'target_sensor'
 CONF_MIN_TEMP = 'min_temp'
 CONF_MAX_TEMP = 'max_temp'
 CONF_TARGET_TEMP = 'target_temp'
-CONF_AC_MODE = 'ac_mode'
+CONF_TARGET_TEMP_HIGH = 'target_temp_high'
+CONF_TARGET_TEMP_LOW = 'target_temp_low'
+CONF_AC_CONTROL = 'ac_control'
 CONF_MIN_DUR = 'min_cycle_duration'
 CONF_COLD_TOLERANCE = 'cold_tolerance'
 CONF_HOT_TOLERANCE = 'hot_tolerance'
 CONF_KEEP_ALIVE = 'keep_alive'
 CONF_INITIAL_OPERATION_MODE = 'initial_operation_mode'
-CONF_AWAY_TEMP = 'away_temp'
+CONF_AWAY_TEMP_COOL = 'away_temp_cool'
+CONF_AWAY_TEMP_HEAT = 'away_temp_heat'
 SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_AWAY_MODE |
```

Comment:
I think this should be built dynamically - e.g., `SUPPORT_TARGET_TEMPERATURE_HIGH` is only true if an ac-entity is set.

Reply:
I think SUPPORT_TARGET_TEMPERATURE_HIGH and SUPPORT_TARGET_TEMPERATURE_LOW must always be set at the same time because you need both heating and cooling to allow Auto. I propose to use
```
        if self.ac_entity_id and self.heater_entity_id:
            self._support_flags = SUPPORT_FLAGS | SUPPORT_TARGET_TEMPERATURE_HIGH | SUPPORT_TARGET_TEMPERATURE_LOW
        else:
            self._support_flags = SUPPORT_FLAGS | SUPPORT_TARGET_TEMPERATURE
```